### PR TITLE
Added utils dir to project

### LINKED ISSUE #23: [new] utils directory
Create `utils` directory, where auxiliary code can be put, like some math operations, which are not included in standard C/C++ libraries.

### PR DESCRIPTION
**Pull request by dante7867**
Some functions and structs added for another task just to fill the directory

### CODE REVIEW

File: untitled-sdl-tank-game/utils/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include "utils.h"
+#include "math.h"
+
+int distanceFromPointToLine(Line line, Point point)
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: please change calculation type to `float`

**Review by pietrik123 — CHANGES_REQUESTED** [review]
good work :) please change calculation type `int` -> `float` in all functions and structures

File: untitled-sdl-tank-game/utils/figures.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+struct Point
+{
+	int PosX;
+	int PosY;
+
+	friend std::ostream& operator << (std::ostream& out, Point point);
+};
+
+struct Round
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/figures.h]: please rename to `Circle`

File: untitled-sdl-tank-game/utils/utils.cpp [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
 int distanceFromPointToLine(Line line, Point point)
 {
-	int distance = (line.Acoef * point.PosX + line.Bcoef * point.PosY + line.Ccoef) /
-		sqrt(pow(line.Acoef, 2) + pow(line.Bcoef, 2));
-	return distance;
+	return (line.Acoef * point.PosX + line.Bcoef * point.PosY + line.Ccoef) /
+		sqrt( pow(line.Acoef, 2) + pow(line.Bcoef, 2) );
 }
 
 bool isRoundAndLineIntersecting(Line line, Round round)
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: please rename to `isCircleAndLineIntersecting`

File: untitled-sdl-tank-game/utils/utils.cpp [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
 {
-	int lineToRoundCenterDistance = distanceFromPointToLine(line, round.center);
-	if (lineToRoundCenterDistance > round.radius)
+	if(distanceFromPointToLine(line, round.center) > round.radius)
 		return false;
 	else
 		return true;
+}
+
+double coutDelta(double a, double b, double c)
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: please rename to `getDelta()`

File: untitled-sdl-tank-game/utils/utils.cpp [hunk 5]
```diff
@@ -1,18 +1,31 @@
...
+{
+	// a*(x^2) + bx + c = 0
+	if( a != 0 && c != 0 )
+		return pow(b , 2) - 4 * a * c;
+}
+
+std::tuple<double, double> countSolutions(double a, double b, double c, double delta)
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: good idea :+1:

File: untitled-sdl-tank-game/utils/utils.cpp [hunk 6]
```diff
@@ -1,18 +1,31 @@
...
+{
+	if( a != 0 )
+		return std::make_tuple<double, double> ( (pow(b, 2) - sqrt(delta)) / (2 * a), ((pow(b, 2) + sqrt(delta)) / (2 * a)) );
```
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: what if `a == 0` ?

**Review by pietrik123 — CHANGES_REQUESTED** [review]
please check comments (renaming proposals mostly). Good job :+1:

File: untitled-sdl-tank-game/utils/utils.cpp (hunk 1 above)
Comment (pietrik123) [inline untitled-sdl-tank-game/utils/utils.cpp]: Optionally (may be in different pull request) please add  some dummy unit tests by:
* adding dependent [project](https://github.com/pietrik123/untitled-sdl-tank-game/wiki/Add-a-project-to-a-Solution-in-MS-Visual) to repo (like [this)](https://github.com/pietrik123/untitled-sdl-tank-game/commit/9d5a40b80e7a3a4fc74982d723bea499bb6f6030#diff-900bcabb885a4243a6ce2c75b247bcec)
* adding *.cpp test file ([example](https://github.com/pietrik123/untitled-sdl-tank-game/blob/9d5a40b80e7a3a4fc74982d723bea499bb6f6030/test_dep_proj2/dummy_test.cpp))

**Review by pietrik123 — APPROVED** [review]
good job, please merge and close task :+1:

**Comment by dantefkr** [comment]
can't create pull request:
### This branch has conflicts that must be resolved
Only those with [write access](https://help.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.

**Pull request merged**